src/main/java/fr/darkvodou/NKtrader/cmds/addTraderCmd.java

```diff
@@ -30,27 +50,311 @@ public boolean onCommand(CommandSender sender, org.bukkit.command.Command comman
+				if(args.length > wPlace + 5)
+				{
+					if(args[wPlace + 5].equals("-n") && args[wPlace + 5].equals("-t"))
+					{
+						sender.sendMessage(ERROR_BAD_ARGS + " 5:\n" + NKT_USAGE_ADDTRADER);
+
+						return true;
+					}
+				}
+
+				if(Bukkit.getWorld(args[wPlace + 1]) == null)
+				{
+					sender.sendMessage(ERROR_WORLD_NOT_EXIST + "");
+
+					return true;
+				}
+
+				world = Bukkit.getWorld(args[wPlace + 1]);
+
+				for(int i = 2; i < 5; i++)
+				{
+					if(!isNumber(args[wPlace + i]))
+					{
+						sender.sendMessage(ERROR_BAD_ARGS + " 6:\n" + NKT_USAGE_ADDTRADER);
+
+						return true;
+					}
+				}
+
+				location = new Location(world, parseDouble(args[wPlace + 2]), parseDouble(args[wPlace + 3]), parseDouble(args[wPlace + 4]));
+			}
+			else
+			{
+				location = ((Player) sender).getLocation();
+			}
+
+			if(manager.getTraderManager().hasTrader(location))
+			{
+				sender.sendMessage(ERROR_TRADER_IS_SET + "");
+				return true;
+			}
+
+			if(nPlace != -1)
+			{
+				if(args.length <= nPlace + 1)
+				{
+					sender.sendMessage(ERROR_BAD_ARGS + " 7:\n" + NKT_USAGE_ADDTRADER);
+
+					return true;
+				}
+
+				int nbArgsForN = 1;
+
+				int j = 0;
+				int i;
+
+				//If name composed increase nbArgs by the length of the composed name
+				if(args[nPlace + 1].charAt(0) == '"')
+				{
+
+					for(i = nPlace + 1; i < args.length && args[i].charAt(args[i].length() - 1) != '"'; i++)
+					{
+						j++;
+					}
+
+					if(args[i].charAt(args[i].length() - 1) != '"')
+					{
+						sender.sendMessage(ERROR_BAD_ARGS + " 8:\n" + NKT_USAGE_ADDTRADER);
+
+						return true;
+					}
+
+					nbArgsForN = j;
+				}
+
+				//If more args after -n args
+				if(args.length > nPlace + nbArgsForN + 2)
+				{
+					if(!args[nPlace + nbArgsForN + 2].equals("-w") && !args[nPlace + nbArgsForN + 2].equals("-t"))
+					{
+						sender.sendMessage(args[nPlace + nbArgsForN + 1]);
+						sender.sendMessage(ERROR_BAD_ARGS + " 9:\n" + NKT_USAGE_ADDTRADER);
+
+						return true;
+					}
+				}
+
+				StringBuilder stringBuilder = new StringBuilder();
+
+				for(i = nPlace + 1; i <= nPlace + j + 1; i++)
+				{
+					stringBuilder.append(args[i]);
+
+					if(i != nPlace + j + 1)
+					{
+						stringBuilder.append(" ");
+					}
+				}
+
+				name = stringBuilder.toString();
+				name = name.replace("\"", "");
+
+				if(name.length() > 100)
+				{
+					sender.sendMessage(ERROR_NAME_LENGTH + "");
+
+					return true;
+				}
+
+				name = name.replace("&", "§");
+
+			}
+
+			if(tPlace != -1)
 			{
-				if(args[0].equals("entity"))
+				if(args.length <= tPlace + 1)
+				{
+					sender.sendMessage(ERROR_BAD_ARGS + " 10:\n" + NKT_USAGE_ADDTRADER);
+
+					return true;
+				}
+
+				//If more args after -t args
+				if(args.length > tPlace + 2)
+				{
+					if(!args[tPlace + 2].equals("-n") && !args[tPlace + 2].equals("-w"))
+					{
+						sender.sendMessage(ERROR_BAD_ARGS + " 11:\n" + NKT_USAGE_ADDTRADER);
+
+						return true;
+					}
+				}
+
```

Comment:
verify type here

Reply:
moved verification in TraderManager.createTraderEntity(trader) or TraderManager.createTraderBlock(trader)